core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/leonardo/LeonardoService.scala

```diff
@@ -121,7 +128,38 @@ class LeonardoResourceDeletionAction(leonardoDAO: LeonardoDAO)(implicit
       }
     }
 
-}
+  /**
+   * Notifies leonardo that it should delete any resource records related to the given google project ID *without*
+   * deleting the actual cloud resources.
+   *
+   * NB: This should only be called after a workspace's google project has been deleted and we want to ensure there
+   * are no further dangling references from Leo to resources within that google project.
+   *
+   * @param googleProjectId ID of the workspace's google project
+   * @param workspaceId ID of the workspace
+   * @param ctx RawlsRequestContext containing auth info
+   */
+  def cleanupResources(googleProjectId: GoogleProjectId, workspaceId: UUID, ctx: RawlsRequestContext)(implicit
+    ec: ExecutionContext
+  ): Future[Unit] =
+    retry(when500OrProcessingException) { () =>
+      Future {
+        blocking {
+          logger.info(
+            s"Sending resource cleanup request to Leonardo [workspaceId=$workspaceId, googleProjectId=${googleProjectId.value}]"
+          )
+          leonardoDAO.cleanupAllResources(ctx.userInfo.accessToken.token, googleProjectId)
+        }
+      }.recoverWith { case t: ApiException =>
+        if (t.getCode != StatusCodes.NotFound.intValue) {
+          logger.warn(
+            s"Unexpected failure cleaning up leonardo workspace resources for workspaceId=$workspaceId . Received ${t.getCode}: [${t.getResponseBody}]"
+          )
+          Future.failed(t)
+        } else {
+          Future.successful()
+        }
+      }
+    }
 
-class LeonardoOperationFailureException(message: String, val workspaceId: UUID)
```

Comment:
This wasn't being used anywhere outside of a test. I'd like to rethink the overall approach in this class to returning the bare Leo client ApiException rather than a higher level exception like this as I think there's real utility there (at the very least, it'd prevent mixing up wsm.ApiException, bpm.ApiException etc. as I often do.)